wolfcrypt/src/port/maxim/maxq10xx.c

```diff
@@ -0,0 +1,3352 @@
+        if (tls13_server_hs_key_obj_id != -1) {
+            key_id = tls13_server_hs_key_obj_id;
+        }
+        else if (tls13_server_app_key_obj_id != -1) {
+            key_id = tls13_server_app_key_obj_id;
+        }
+        else {
+            WOLFSSL_MSG("MAXQ: tls record decryption key is not selected");
+        }
+    }
+    mxq_algo_id_t algo_id = 0;
+
+    if (ssl->specs.bulk_cipher_algorithm == wolfssl_aes_gcm) {
+        algo_id = ALGO_CIPHER_AES_GCM;
+    } else if (ssl->specs.bulk_cipher_algorithm == wolfssl_aes_ccm) {
+        algo_id = ALGO_CIPHER_AES_CCM;
+    }
+
+    rc = wolfSSL_CryptHwMutexLock();
+    if (rc != 0) {
+        return rc;
+    }
+
+    WOLFSSL_MSG("MAXQ: MXQ_TLS13_Update_IV()");
+    mxq_rc = MXQ_TLS13_Update_IV( key_id, (mxq_u1 *)iv, ivSz);
+    if (mxq_rc) {
+        WOLFSSL_MSG("MAXQ: MXQ_TLS13_Update_IV() failed");
+        wolfSSL_CryptHwMutexUnLock();
+        return WC_HW_E;
+    }
+
+    mxq_rc = maxq10xx_cipher_do(algo_id, is_encrypt, key_id, (mxq_u1 *)in, out,
+                                sz, (mxq_u1 *)iv, ivSz, (mxq_u1 *)authIn,
+                                authInSz, authTag, authTagSz);
+    if (mxq_rc) {
+        WOLFSSL_MSG("MAXQ: maxq10xx_cipher_do() failed");
+        wolfSSL_CryptHwMutexUnLock();
+        return WC_HW_E;
+    }
+
+    wolfSSL_CryptHwMutexUnLock();
+    return 0;
+}
+#endif /* HAVE_HKDF */
+#endif /* HAVE_PK_CALLBACKS && WOLFSSL_MAXQ108x */
+
+void maxq10xx_SetupPkCallbacks(struct WOLFSSL_CTX* ctx, ProtocolVersion *pv)
+{
+    (void)pv;
+
+    if (pv == NULL) {
+        return;
+    }
+
+    WOLFSSL_ENTER("maxq10xx_SetupPkCallbacks");
+    if (init_pk_callbacks) {
+        return;
+    }
+
+#ifdef WOLFSSL_MAXQ108x
+    #ifdef HAVE_HKDF
+    wolfSSL_CTX_SetHKDFExtractCb(ctx, crypto_hkdf_extract);
+    wolfSSL_CTX_SetHKDFExpandLabelCb(ctx, maxq10xx_HkdfExpandLabel);
+    use_hw_hkdf_expand = 1;
+    #endif
+
+    if ((pv->major == SSLv3_MAJOR) && (pv->minor == WOLFSSL_TLSV1_3)) {
+        tls13active = 1;
+        wolfSSL_CTX_SetEccKeyGenCb(ctx, maxq10xx_create_ecc_key_cb);
+        wolfSSL_CTX_SetEccSharedSecretCb(ctx, maxq10xx_shared_secret_cb);
+        wolfSSL_CTX_SetDhGenerateKeyPair(ctx, maxq10xx_DhGenerateKeyPair);
+        wolfSSL_CTX_SetDhAgreeCb(ctx, maxq10xx_DhAgreeCb);
+        wolfSSL_CTX_SetEccVerifyCb(ctx, maxq10xx_verify_signature_cb);
+        wolfSSL_CTX_SetRsaPssSignCb(ctx, maxq10xx_RsaPssSign);
+        wolfSSL_CTX_SetRsaPssSignCheckCb(ctx, maxq10xx_SkipSigCheck);
+        wolfSSL_CTX_SetRsaPssVerifyCb(ctx, maxq10xx_RsaPssVerify);
+
+        wolfSSL_CTX_SetPerformTlsRecordProcessingCb(ctx,
+            maxq10xx_perform_tls13_record_processing);
+
+    } else
+#endif /* WOLFSSL_MAXQ108x */
+    {
+        wolfSSL_CTX_SetEccKeyGenCb(ctx, maxq10xx_perform_client_key_exchange);
+        wolfSSL_CTX_SetPerformTlsRecordProcessingCb(ctx,
+            maxq10xx_perform_tls12_record_processing);
+    }
+
+    wolfSSL_CTX_SetProcessServerCertCb(ctx,
```

Comment:
This callback needs to be renamed.  `Server` ----> `Peer` .    All associated pointers and types and functions need to be renamed as well.

Reply:
Done in 3b5a8330c